sklearn/model_selection/_split.py

```diff
@@ -637,6 +637,121 @@ def split(self, X, y, labels=None):
         """
         return super(StratifiedKFold, self).split(X, y, labels)
 
+class HomogeneousTimeSeriesCV(_BaseKFold):
+    """Homogeneous Time Series cross-validator
+
+    Provides train/test indices to split time series data in train/test sets.
+
+    This cross-validation object is a variation of KFold.
+    In iteration k, it returns first k folds as train set and k+1 fold as
+    test set.
+
+    Read more in the :ref:`User Guide <cross_validation>`.
+
+    Parameters
+    ----------
+    n_folds : int, default=3
+        Number of folds. Must be at least 2.
+
+    Examples
+    --------
+    >>> from sklearn.model_selection import HomogeneousTimeSeriesCV
+    >>> X = np.array([[1, 2], [3, 4], [1, 2], [3, 4]])
+    >>> y = np.array([1, 2, 3, 4])
+    >>> htscv = HomogeneousTimeSeriesCV(n_folds=4)
```

Comment:
It is also mentioned in following blog:

> Split data in train and test set given a Date (i.e. test set is what happens after 2 April 2014 included).
Split train set (i.e. what happens before 2 April 2014 not included) in for example 10 consecutive time folds.
Then, in order not lo lose the time information, perform the following steps:
Train on fold 1 –>  Test on fold 2
Train on fold 1+2 –>  Test on fold 3
Train on fold 1+2+3 –>  Test on fold 4
Train on fold 1+2+3+4 –>  Test on fold 5
Train on fold 1+2+3+4+5 –>  Test on fold 6
Train on fold 1+2+3+4+5+6 –>  Test on fold 7
Train on fold 1+2+3+4+5+6+7 –>  Test on fold 8
Train on fold 1+2+3+4+5+6+7+8 –>  Test on fold 9
Train on fold 1+2+3+4+5+6+7+8+9 –>  Test on fold 10
Compute the average of the accuracies of the 9 test folds (number of folds  – 1)

http://francescopochetti.com/pythonic-cross-validation-time-series-pandas-scikit-learn/